src/bin/jp2/opj_decompress.c

```diff
@@ -475,7 +475,7 @@ const char* path_separator = "/";
 #endif
 
 /* -------------------------------------------------------------------------- */
-char get_next_file(int imageno, dircnt_t *dirptr, img_fol_t *img_fol,
+char get_next_file(unsigned int imageno, dircnt_t *dirptr, img_fol_t *img_fol,
```

Comment:
fprintf(stderr, "File Number %d \"%s\"\n", imageno, image_filename); at line 486 should use "%u" instead of "%d". There might be other similar instances in the other files